--- /usr/sbin/fence_azure_arm    2021-04-30 03:58:28.729002075 +0000
+++ /usr/sbin/fence_azure_arm.new    2021-04-30 03:58:59.821059338 +0000
@@ -115,7 +115,7 @@
 
         if (options["--action"]=="off"):
             logging.info("Poweroff " + vmName + " in resource group " + rgName)
-            compute_client.virtual_machines.power_off(rgName, vmName, skip_shutdown=True)
+            compute_client.virtual_machines.begin_power_off(rgName, vmName, skip_shutdown=True)
         elif (options["--action"]=="on"):
             logging.info("Starting " + vmName + " in resource group " + rgName)
             compute_client.virtual_machines.start(rgName, vmName)
